docs/src/developer/reference/oauth.md

```diff
@@ -0,0 +1,402 @@
+| --------------- | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `scope`         | Required. The scope of the access request.                                                                                                                                                                                                                                        |
+| `response_type` | Required. Value MUST be set to "code".                                                                                                                                                                                                                                            |
+| `client_id`     | Required. The client identifier.                                                                                                                                                                                                                                                  |
+| `redirect_url`  | Required. MUST match the URL that was provided during client registration                                                                                                                                                                                                         |
+| `state`         | Required. An opaque value used by the client to maintain state between the request and callback.</br>The authorization server includes this value when redirecting the user-agent back to the client.</br>The parameter SHOULD be used for preventing cross-site request forgery. |
+
+Once the user consents the user agent will be redirected back to the 3rd party app, using the redirect URI provided during client registration, with an authorization code which can be used to retrieve an access token and a refresh token and is good for one use.
+
+Example response:
+
+```http
+HTTP/1.1 302 Found
+Transfer-Encoding: chunked
+Date: Thu, 23 Feb 2023 15:50:21 GMT
+Server: Warp/3.3.23
+Location: https://example.com?code=1395a1a44b72e0b81ec8fe6c791d2d3f22bc1c4df96857a88c3e2914bb687b7b&state=foobar
+Vary: Accept-Encoding
+```
+
+### Retrieve access and refresh token
+
+The 3rd party app sends the authorization code together with the client credentials and the parameters shown below using the `application/x-www-form-urlencoded` format with character encoding of UTF-8 to the authorization server to retrieve an access token and a refresh token:
+
+```curl
+curl -s -X POST localhost:8080/oauth/token \
```

Comment:
I had to look twice to understand this is the "external" endpoint of wire-server (the internal uses port 8082)

The RFC uses server.example.com and client.example.com. Maybe we can use the same consistently too? (and use server-internal.example.com for the internal endpoint)